give up on two way binding to prevent empty calendar

ulferts opened the pull request:
https://community.openproject.com/projects/openproject/work_packages/29117

ulferts committed:
give up on two way binding to prevent empty calendar

oliverguenther commented on frontend/src/app/components/wp-calendar/wp-calendar.component.ts:
```diff
@@ -159,6 +158,15 @@ export class WorkPackagesCalendarController implements OnInit, OnDestroy {
+    // in a grid, having two way binding will lead to having constantly
+    // removed the events after showing them initially.
+    // It appears as if the two way binding is initialized twice if used.
+    setTimeout(() => {
```
What do you need the timeout for?

ulferts replied:
The fix is basically a rebuild of the two way binding (without the two way binding) for setting events as done in the [ng-fullcalendar npm package](https://github.com/Jamaks/ng-fullcalendar/blob/master/src/lib/src/calendar.component.ts#L26). The two way binding does not work as the assignment seems to be called twice (once with the desired results and once with undefined). 
Without the timeout
```
TypeError: Cannot read property 'renderEvents' of undefined
```
is thrown for whatever reasons.

oliverguenther replied:
That's due to the calendar not yet being initialized. I allowed myself to push a commit that avoids using the timeout, and instead uses a callback from ng-fullcalendar to set up the WP results listener

oliverguenther committed:
Use ng-fullcalendar initialized callback to set up listener

When setting up the WP listener in `ngOnInit`, the children are not
yet initialized. If the results are in faster than the calendar
component is initialized, then an error is emitted because it's simply
not there yet.

We can use the initialized emitter from ng-fullcalendar to ensure it's
there
